File: Sources/CxxSPM/ImageProcessor.swift
import CxxLib

public class ImageProcessorSwift {
    private var processor: ImageProcessor!
    
    public init() {
        processor = ImageProcessor()
    }

    public func loadImage(_ path: String) -> Bool {
        return processor.loadImage(std.string(path))
    }
    
    public func saveImage(_ path: String) -> Bool {
        return processor.saveImage(std.string(path))
    }
    
    public func convertToGrayscale() {
        return processor.convertToGrayscale()
    }
    
    public func blur(_ kernelSize: Int32) {
        return processor.blur(Int32(kernelSize))
    }
    
    public func resize(_ width: Int32, _ height: Int32) {
        return processor.resize(width, height)
    }
}

File: Sources/CxxLib/ImageProcessor.cpp
#include "include/ImageProcessor.hpp"

ImageProcessor::ImageProcessor() {}

bool ImageProcessor::loadImage(const std::string& path) {
    image = cv::imread(path);
    return !image.empty();
}

bool ImageProcessor::saveImage(const std::string& path) {
    return cv::imwrite(path, image);
}

void ImageProcessor::convertToGrayscale() {
    if (!image.empty()) {
        cv::cvtColor(image, image, cv::COLOR_BGR2GRAY);
    }
}

void ImageProcessor::blur(int kernelSize) {
    if (!image.empty()) {
        cv::GaussianBlur(image, image, cv::Size(kernelSize, kernelSize), 0);
    }
}

void ImageProcessor::resize(int width, int height) {
    if (!image.empty()) {
        cv::resize(image, image, cv::Size(width, height));
    }
} 
